srt-protocol/src/protocol/pending_connection/listen.rs

```diff
@@ -182,42 +225,105 @@ impl Listen {
         }
     }
 
-    fn handle_control_packets(
+    fn request_access(
         &mut self,
-        control: ControlPacket,
-        from: SocketAddr,
-        now: Instant,
-        acceptor: &mut impl StreamAcceptor,
+        remote: SocketAddr,
+        local_socket_id: SocketId,
+        timestamp: TimeStamp,
+        state: ConclusionWaitState,
+        shake: HandshakeControlInfo,
+        incoming: HsV5Info,
     ) -> ConnectionResult {
-        match (self.state.clone(), control.control_type) {
-            (InductionWait, ControlTypes::Handshake(shake)) => {
-                self.wait_for_induction(from, control.timestamp, shake, now)
-            }
-            (ConclusionWait(state), ControlTypes::Handshake(shake)) => {
-                self.wait_for_conclusion(from, control.timestamp, state, shake, now, acceptor)
-            }
-            (InductionWait, control_type) | (ConclusionWait(_), control_type) => {
-                NotHandled(ConnectError::HandshakeExpected(control_type))
-            }
-        }
+        // TODO: handle StreamId parsing error
+        let sid = incoming.sid.clone().unwrap_or_default().into_bytes();
+        let stream_id = StreamId::try_from(sid).unwrap();
```

Comment:
This also seems to duplicate the work of validating that a StreamId is UTF-8--maybe StreamId should `impl From<String>`?

Reply:
Yeah, perhaps. The goal was to, ergonomically, ensure all StreamIds instances are never in an invalid state (i.e. longer than 512 characters), I'm open to any improvements towards that goal.

Reply:
Ok, I actually implemented this in the separate "options" PR. I'll probably have to fix this when I merge/rebase.